### PR TITLE
Block graph management

### PR DESCRIPTION
**Pull request by pingpongsneak**

### CODE REVIEW

File: cypcore/Ledger/Graph.cs [hunk 1]
```diff
@@ -150,6 +151,9 @@ private IDisposable BlockmaniaListener()
                     {
                         var blockGraphs = _unitOfWork.BlockGraphRepository
                             .WhereAsync(x => new ValueTask<bool>(x.Block.Round == NextRound())).AsTask().Result;
+
+                        if (blockGraphs.Count < 2) return;
```
Comment (sanastasiou) [inline cypcore/Ledger/Graph.cs:155]: Please explain the purpose of this.

File: cypcore/Consensus/Blockmania.cs [hunk 2]
```diff
@@ -762,13 +762,7 @@ private async Task Run(ChannelReader<BlockGraph> reader)
...
         {
             _logger.Here().Debug("Adding block to graph block.id: {@BlockId}", data.Block);
-
-            var task = Task.Factory.StartNew(async () =>
-            {
-                await _entries.Writer.WriteAsync(data);
-            });
-
-            task.Wait();
+            Task.Factory.StartNew(async () => { await _entries.Writer.WriteAsync(data); });
```
Comment (sanastasiou) [inline cypcore/Consensus/Blockmania.cs:765]: So this thread runs forever and cannot be gracefully cancelled?

**Review by sanastasiou — APPROVED** [review]

File: cypcore/Consensus/Blockmania.cs (hunk 2 above)
Comment (pingpongsneak) [reply cypcore/Consensus/Blockmania.cs:765]: Same as with a blocking thread no?

File: cypcore/Ledger/Graph.cs (hunk 1 above)
Comment (pingpongsneak) [reply cypcore/Ledger/Graph.cs:155]: I sure its self explanatory what the short circuit statement is doing?

**Pull request merged**